File: Algorithms/QuickSort.java
//QuickSort in Java
class quickSort {

    // Swap two elements
    static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    // Taking last element as pivot and sorting
    static int partition(int[] arr, int low, int high) {
        int pivot = arr[high];

        int i = (low - 1);

        for (int j = low; j <= high - 1; j++) {
            if (arr[j] < pivot) {
                i++;
                swap(arr, i, j);
            }
        }
        swap(arr, i + 1, high);
        return (i + 1);
    }

    static void qSort(int[] arr, int low, int high) {
        if (low < high) {

            int pi = partition(arr, low, high);

            qSort(arr, low, pi - 1);
            qSort(arr, pi + 1, high);
        }
    }

    static void printArray(int[] arr, int size) {
        for (int i = 0; i < size; i++)
            System.out.print(arr[i] + " ");

        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9};
        int n = arr.length;

        qSort(arr, 0, n - 1);
        System.out.println("Sorted array: ");
        printArray(arr, n);
    }
}

File: Algorithms/quicksort.py
#QUICK SORT

from sys import stdin, stdout

# This function takes first element as pivot, places the pivot element at its correct position in sorted array,
# and places all smaller (smaller than pivot) to left of pivot and all greater elements to right of pivot


def partition(ar, low, high):

    pivot = ar[low]
    i, j = low, high

    while (i < j):
        # loop runs until we receive element>pivot
        while (i <= high and ar[i] <= pivot):
            i += 1
        # loop runs until we receive element<pivot
        while (j >= 0 and ar[j] > pivot):
            j -= 1

        if(i < j):
            ar[i], ar[j] = ar[j], ar[i]

    ar[low], ar[j] = ar[j], ar[low]
    return(j)


def QuickSort(ar, low, high):
    if(len(ar) == 1):
        return (ar)
    if(low < high):
        j = partition(ar, low, high)
        QuickSort(ar, low, j-1)
        QuickSort(ar, j+1, high)


if __name__ == "__main__":
    ar = list(map(int, stdin.readline().split()))
    low, high = 0, len(ar)-1
    QuickSort(ar, low, high)
    print(*ar)